hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/ExpressionUtil.java

```diff
@@ -59,41 +58,38 @@ private ExpressionUtil() {
             for (FieldCollation fieldCollation : fieldCollationList) {
                 // For each collation:
                 // - Get collation index and use it to fetch values from the rows.
-                // - Get direction (ASCENDING, DESCENDING) and null direction
-                //   (NULLS FIRST, NULLS LAST). If no null direction is given, then
-                //   it will be inferred from the direction. Since NULL is sorted as
-                //   the +Inf, ASCENDING implies NULL LAST whereas DESCENDING implies
-                //   NULLS FIRST.
+                // - Get direction (ASCENDING, DESCENDING)
                 // - Comparison of field values:
-                //   - If both of them are NULL, then return 0.
+                //   - If both of them are NULL, then result is 0.
                 //   - Otherwise, if one of them is NULL, then return:
-                //     - null direction value if LHS is NULL.
-                //     - or negative null direction if RHS is NULL.
-                //   - If none of them is NULL, then:
-                //     - If direction is ASCENDING, then return the comparison result.
-                //     - If direction is DESCENDING, return the negation of comparison result.
+                //     - result is -1 if LHS is NULL.
+                //     - result is 1 if RHS is NULL.
+                // - Return the result if ASCENDING
+                //   Return the reverted result if DESCENDING
                 int index = fieldCollation.getIndex();
 
                 Comparable o1 = (Comparable) row1[index];
                 Object o2 = row2[index];
 
                 Direction direction = fieldCollation.getDirection();
 
-                NullDirection nullDirection = fieldCollation.getNullDirection();
-                if (nullDirection == null) {
-                    nullDirection = direction.defaultNullDirection();
-                }
-
                 int result;
                 if (o1 == null && o2 == null) {
                     result = 0;
                 } else if (o1 == null) {
-                    result = nullDirection.nullComparison;
+                    result = -1;
                 } else if (o2 == null) {
-                    result = -nullDirection.nullComparison;
+                    result = 1;
                 } else {
                     result = o1.compareTo(o2);
-                    result = direction.isDescending() ? -result : result;
+                }
+
+                if (direction.isDescending()) {
+                    if (result < 0) {
```

Comment:
Wouldn't a plain  `result = -result` work here too?

Reply:
https://github.com/hazelcast/hazelcast/pull/19014#discussion_r666078397

Check out this comment. It is highly unlikely occurrence (standard types for my JDK uses -1, 0, 1 as only values) but a valid concern.